x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/source_logic.ts

```diff
@@ -155,8 +155,14 @@ export const SourceLogic = kea<MakeLogicType<SourceValues, SourceActions>>({
           clearFlashMessages();
         }
       } catch (e) {
-        if (e.response.status === 404) {
-          KibanaLogic.values.navigateToUrl(NOT_FOUND_PATH);
+        if (e?.response?.status === 404) {
+          const redirect = isOrganization ? SOURCES_PATH : PERSONAL_SOURCES_PATH;
+          KibanaLogic.values.navigateToUrl(redirect);
+          setErrorMessage(
+            i18n.translate('xpack.enterpriseSearch.workplaceSearch.sources.notFoundErrorMessage', {
+              defaultMessage: 'Source not found.',
+            })
+          );
```

Comment:
Screencap of this UX:

![source_404](https://user-images.githubusercontent.com/549407/123198704-703f1d00-d462-11eb-9885-0153c3223b9e.gif)

Screencap of this UX on personal dashboards:

![source_404_p](https://user-images.githubusercontent.com/549407/123198724-733a0d80-d462-11eb-854a-66d0c555b53a.gif)

Previously this was sending personal dashboard source 404s to the organization dashboard, which we didn't want.